internal/apps/devflow/flow/flow.service.go

```diff
@@ -577,8 +577,10 @@ func (s *Service) GetDevFlowInfo(ctx context.Context, req *pb.GetDevFlowInfoRequ
 		work.AddFunc(func(locker *limit_sync_group.Locker, i ...interface{}) error {
 			appID := i[0].(uint64)
 			appDto := appInfoMap[appID]
+			locker.Lock()
 			appTempBranchChangeBranchListMap[appID] = make(map[string][]*pb.ChangeBranch)
 			appTempBranchCommitMap[appID] = make(map[string]*apistructs.Commit)
+			locker.Unlock()
```

Comment:
the locked scope is too small.
`appTempBranchChangeBranchListMap` can be overwritten in L584 by another async func.

Reply:
appTempBranchChangeBranchListMap inL584 already locked